### ISSUE TITLE
No notificatons in Ubuntu

### ISSUE THREAD

**Issue by steel-sword** [laurent22/joplin#6029]
## Environment

Joplin version: v2.6.10
Platform: Linux
OS specifics: Ubuntu 20, Joplin Installed from Snap

## Steps to reproduce

1. Created a new to-do note.
2. Wrote its name.
3. Put example to do by writing `- [ ] some to do`
4. WISIWIG has parsed it and showed me a checkbox.
5. Then I pressed "Set alarm" on top of to-do note.
6. Chose 2 minutes after current time.
7. Waited two minutes.
8. Nothing happened

## Describe what you expected to happen
I expected a notification but found nothing.

## Logfile

```
11:49:00: AlarmServiceDriverNode::scheduleNotification: Triggering notification (default): 
{appID: 'net.cozic.joplin-desktop', title: 'To do', icon: '/snap/joplin-desktop/23/opt/joplin-desktop/resources/build/icons/512x512.png', message: '- [ ]  plans for today'}
appID: "net.cozic.joplin-desktop"
icon: "/snap/joplin-desktop/23/opt/joplin-desktop/resources/build/icons/512x512.png"
message: "- [ ]  plans for today"
title: "To do"
[[Prototype]]: Object
Logger.ts:217 11:49:00: AlarmServiceDriverNode::scheduleNotification: node-notifier response: Error: Command failed: notify-send "To do" "- [ ]  plans for today" --icon "/snap/joplin-desktop/23/opt/joplin-desktop/resources/build/icons/512x512.png" --expire-time "10000"
Unknown option - [ ]  plans for today

    at ChildProcess.exithandler (child_process.js:328)
    at ChildProcess.emit (events.js:376)
    at maybeClose (internal/child_process.js:1055)
    at Socket.<anonymous> (internal/child_process.js:441)
    at Socket.emit (events.js:376)
    at Pipe.<anonymous> (net.js:673) undefined
```

**Comment by MrCarroll** [comment]
You can replicate this error without Joplin, the problem is the syntax of the `notify-send` command being generated.

`notify-send "To do" "- [ ]  plans for today" --icon "/snap/joplin-desktop/23/opt/joplin-desktop/resources/build/icons/512x512.png" --expire-time "10000"`

If you remove `- [ ] ` it works, I wonder if Joplin is substituting this instead of an Emoji or something similar?

**Comment by roman-r-m** [comment]
> If you remove `- [ ] ` it works, I wonder if Joplin is substituting this instead of an Emoji or something similar?

It's an unchecked checkbox in markdown. Should probably be escaped.

**Comment by steel-sword** [comment]
@MrCarroll, yes, I understood that it's from `notify-send`. I put some text before `- [ ]` and it worked properly. But it's a bug anyway, isn't it?

**Comment by MrCarroll** [comment]
Sure, it's still a bug; but I was just making sure it was identified as a problem with the `notify-send` binary and not the fact it's in a Snap package, because the Snap package is strictly speaking unsupported because it's a third party package.

Me and my friend had a long think about the semantics going on here, and the fix is actually rather simple, you need to pass the arguments first and the input last; seperating the two with `--`, e.g

`notify-send --icon "/snap/joplin-desktop/23/opt/joplin-desktop/resources/build/icons/512x512.png" --expire-time "10000" -- "To do" "- [ ] plans for today"`

However I haven't a clue where in the codebase this is actually handled, but passing `--` to seperate arguements from input is cleaner than e.g escaping the input.

**Comment by roman-r-m** [comment]
Looks to be using `node-notifier` lib here: https://github.com/laurent22/joplin/blob/dev/packages/lib/services/AlarmServiceDriverNode.ts

**Comment by ManavSarkar** [comment]
> Sure, it's still a bug; but I was just making sure it was identified as a problem with the `notify-send` binary and not the fact it's in a Snap package, because the Snap package is strictly speaking unsupported because it's a third party package.
> 
> Me and my friend had a long think about the semantics going on here, and the fix is actually rather simple, you need to pass the arguments first and the input last; separating the two with `--`, e.g
> 
> `notify-send --icon "/snap/joplin-desktop/23/opt/joplin-desktop/resources/build/icons/512x512.png" --expire-time "10000" -- "To do" "- [ ] plans for today"`
> 
> However I haven't a clue where in the codebase this is actually handled, but passing `--` to separate arguments from input is cleaner than e.g escaping the input.

But for that we need to modify the node-notifier package itself, and then update it here, isn't it

**Comment by ManavSarkar** [comment]
> > Sure, it's still a bug; but I was just making sure it was identified as a problem with the `notify-send` binary and not the fact it's in a Snap package, because the Snap package is strictly speaking unsupported because it's a third party package.
> > Me and my friend had a long think about the semantics going on here, and the fix is actually rather simple, you need to pass the arguments first and the input last; separating the two with `--`, e.g
> > `notify-send --icon "/snap/joplin-desktop/23/opt/joplin-desktop/resources/build/icons/512x512.png" --expire-time "10000" -- "To do" "- [ ] plans for today"`
> > However I haven't a clue where in the codebase this is actually handled, but passing `--` to separate arguments from input is cleaner than e.g escaping the input.
> 
> But for that we need to modify the node-notifier package itself, and then update it here, isn't it. And what about trimming any - before the message and title, it wouldn't look good in notification also as a dash before message.

@MrCarroll

**Comment by MrCarroll** [comment]
Yes, ideally this would be fixed upstream and I've created an issue about this so they can consider it.

And yes, it could also be fixed downstream in Joplin, e.g by trimming any leading dashes, but I'm not going to promote that choice unless upstream isn't interested in fixing it for whatever reason. If the user thinks the `-` looks bad, they can remove it themselves. Some users might think the `-` looks good, given they put it there themselves.
